deployment/serverless.yml

```diff
@@ -19,48 +19,91 @@ provider:
     Product: ${env:PROJECT_GROUP_SHORT}
     Programme: ${env:PROGRAMME}
     Service: ${env:PROJECT_GROUP_SHORT}
+  apiGateway:
+    apiKeySourceType: HEADER
+    apiKeys:
+      - name: ${env:NHS_UK_API_KEY_KEY}
+        value: ${self:custom.secret.${env:NHS_UK_API_KEY_KEY}}
+  endpointType: regional
+  tracing:
+    apiGateway: true
+    lambda: true
+
+custom:
+  secret: ${ssm:/aws/reference/secretsmanager/${env:API_GATEWAY_API_KEY_NAME}}
+
+  # iam:
+  #   role:
+  #     name: dos-integration-${sls:stage}-lambda-role
+  #     statements:
+        # - Effect: 'Allow'
+        #   Resource: 'TODO'
+        #   Action: 'TODO' #This need to be restricted when used
+        # - Effect: 'Deny'
+        #   Resource: 'TODO'
+        #   Action: 'TODO'
 
 plugins:
   - serverless-vpc-discovery
   - serverless-localstack
 
-custom:
-  vpcDiscovery:
```

Comment:
Also the other custom section can be merged into this custom

Reply:
Resolved